main.go

```diff
@@ -32,85 +34,139 @@ type MyResponse struct {
+}
+
+func (l Lambda) downloadFile(bucket string, key string) error {
+	filename := l.path()
 
-	filename := "/tmp/file"
 	f, err := os.Create(filename)
 	if err != nil {
 		log.Printf("failed to create file %q, %v", filename, err)
-		return MyResponse{}, fmt.Errorf("failed to create file %q, %v", filename, err)
+		return fmt.Errorf("failed to create file %q, %v", filename, err)
 	}
 
-	_, err = downloader.Download(f, &s3.GetObjectInput{
-		Bucket: aws.String(bucketName),
-		Key:    aws.String(objectKey),
+	_, err = l.downloader.Download(f, &s3.GetObjectInput{
+		Bucket: aws.String(bucket),
+		Key:    aws.String(key),
 	})
 
 	if err != nil {
 		log.Printf("failed to download file, %v", err)
-		return MyResponse{}, fmt.Errorf("failed to download file, %v", err)
+		return fmt.Errorf("failed to download file, %v", err)
 	}
 
-	// scan with clamav
-	status := "ok"
-	cmd := exec.Command("./bin/clamscan", filename)
+	return nil
+}
+
+func (l Lambda) scanFile() (string, error) {
+	status := l.tagValues.pass
+	cmd := exec.Command("./bin/clamscan", l.path())
 	if err := cmd.Run(); err != nil {
 		if exitError, ok := err.(*exec.ExitError); ok {
 			if exitError.ExitCode() == 1 {
-				status = "infected"
+				status = l.tagValues.fail
 			} else {
 				log.Printf("failed to scan file, %v", err)
-				return MyResponse{}, fmt.Errorf("failed to scan file, %v", err)
+				return "", fmt.Errorf("failed to scan file, %v", err)
 			}
 		} else {
 			log.Printf("failed to scan file, %v", err)
-			return MyResponse{}, fmt.Errorf("failed to scan file, %v", err)
+			return "", fmt.Errorf("failed to scan file, %v", err)
 		}
 	}
 
-	// tag document
-	s3client := s3.New(sess)
+	return status, nil
+}
 
-	tagging, err := s3client.GetObjectTagging(&s3.GetObjectTaggingInput{
-		Bucket: aws.String(bucketName),
-		Key:    aws.String(objectKey),
+func (l Lambda) tagFile(bucket string, key string, status string) error {
+	tagging, err := l.s3.GetObjectTagging(&s3.GetObjectTaggingInput{
+		Bucket: aws.String(bucket),
+		Key:    aws.String(key),
 	})
 
 	if err != nil {
 		log.Printf("failed to get tags, %v", err)
-		return MyResponse{}, fmt.Errorf("failed to get tags, %v", err)
+		return fmt.Errorf("failed to get tags, %v", err)
 	}
 
-	_, err = s3client.PutObjectTagging(&s3.PutObjectTaggingInput{
-		Bucket: aws.String(bucketName),
-		Key:    aws.String(objectKey),
+	_, err = l.s3.PutObjectTagging(&s3.PutObjectTaggingInput{
+		Bucket: aws.String(bucket),
+		Key:    aws.String(key),
 		Tagging: &s3.Tagging{
 			TagSet: append(tagging.TagSet, &s3.Tag{
-				Key:   aws.String("virus-scan-status"),
+				Key:   aws.String(l.tagKey),
 				Value: aws.String(status),
 			}),
 		},
 	})
 
 	if err != nil {
 		log.Printf("failed to write tags, %v", err)
-		return MyResponse{}, fmt.Errorf("failed to write tags, %v", err)
+		return fmt.Errorf("failed to write tags, %v", err)
+	}
+
+	return nil
+}
+
+func (l Lambda) HandleEvent(event ObjectCreatedEvent) (MyResponse, error) {
```

Comment:
I'd probably put all these methods on `*Lamdba` instead, then use `&Lambda{}` to create